### ISSUE TITLE
如何跑自己的2维多元csv数据集

### ISSUE THREAD

**Issue by fizzking** [ZZUFaceBookDL/GTN#4]
你的论文和代码跑的是多元的3维时间序列，用的是mat文件格式，如果我希望跑的数据集是2维的多元时间序列，用的是csv文件格式如下图，最后一列vs是标签，我该怎么修改代码，可以讲讲大概思路吗？麻烦作者了。
![image](https://user-images.githubusercontent.com/111410702/229337909-48f72a7d-586b-4184-9ce7-240b1623925a.png)

**Comment by paullintilhac** [comment]
running into the same issue as well. I may create a PR for this in the coming week, assuming I can get the original dataset up and running (currently struggling even with that, since I can't even get access to the dataset on Baidu...)

**Comment by superrCoder** [comment]
> 你的论文和代码跑的是多元的3维时间序列，用的是mat文件格式，如果我希望跑的数据集是2维的多元时间序列，用的是csv文件格式如下图，最后一列vs是标签，我该怎么修改代码，可以讲讲大概思路吗？麻烦作者了。 ![image](https://user-images.githubusercontent.com/111410702/229337909-48f72a7d-586b-4184-9ce7-240b1623925a.png)

请问你现在完成的了吗 我的想法是把csv文件也做成这个mat的格式但是没有成功

**Comment by Lllllyq** [comment]
你可以自己写一个dataloader，用于加载你的csv数据，
![image](https://github.com/ZZUFaceBookDL/GTN/assets/102804756/3b983e3a-0c1c-41fa-b366-24d84d6861ad)，我的数据形式和你一样，一行八个特征，最后一列为标签，读数据时一行一行读。
class CustomDataset(Dataset):
    def __init__(self, csv_file):
        self.data = pd.read_csv(csv_file,header = None)
        self.features = self.data.iloc[:, :8].values
        self.labels = self.data.iloc[:, 8].values

    def __len__(self):
        return len(self.data)

    def __getitem__(self, index):
        features = torch.tensor(self.features[index], dtype=torch.float32)
        features = features.unsqueeze(1)
        label = torch.tensor(self.labels[index], dtype=torch.long)
        return features, label
